src/panels/config/devices/ha-config-devices-dashboard.ts

```diff
@@ -342,6 +489,136 @@ export class HaConfigDeviceDashboard extends LitElement {
     const deviceId = ev.detail.id;
     navigate(this, `/config/devices/device/${deviceId}`);
   }
+
+  private _showDisabledChanged(ev: CustomEvent<RequestSelectedDetail>) {
+    if (ev.detail.source !== "property") {
+      return;
+    }
+    this._showDisabled = ev.detail.selected;
+  }
+
+  private _handleSearchChange(ev: CustomEvent) {
+    this._filter = ev.detail.value;
+  }
+
+  private _clearFilter() {
+    navigate(this, window.location.pathname, true);
+  }
+
+  private _showAll() {
+    this._showDisabled = true;
+  }
+
+  static get styles(): CSSResult[] {
+    return [
+      haStyle,
+      css`
+        hass-loading-screen {
+          --app-header-background-color: var(--sidebar-background-color);
```

Comment:
Copied from /src/panels/config/entities/ha-config-entities.ts, could be moved to a separate component to avoid duplicating the code. Also, not all the styling may be needed.